chain/chain/src/chain.rs

```diff
@@ -3261,9 +3296,13 @@ impl<'a> ChainUpdate<'a> {
             } else {
                 &prev_header
             };
+            let last_final_block = *new_head_header.last_final_block();
 
             let tip = Tip::from_header(new_head_header);
             self.chain_store_update.save_head(&tip)?;
+            let new_final_header =
+                self.chain_store_update.get_block_header(&last_final_block)?.clone();
+            self.chain_store_update.save_final_head(&Tip::from_header(&new_final_header))?;
```

Comment:
Right, but does it really matter? In this case there is an unexpected change to final head anyways, and, since doomslug has liveness, the final head should be updated to the actual final head relatively quickly.